code/modules/mob/living/simple_animal/parrot.dm

```diff
@@ -858,11 +864,20 @@
 	gold_core_spawnable = 0
 	speak_chance = 3
 	var/memory_saved = 0
+	var/rounds_survived = 0
+	var/longest_survival = 0
 
 /mob/living/simple_animal/parrot/Poly/New()
 	ears = new /obj/item/device/radio/headset/headset_eng(src)
 	available_channels = list(":e")
 	Read_Memory()
+	if(rounds_survived == longest_survival)
+		speak += pick("...[longest_survival].", "The things I've seen!", "I have lived many lives!", "What are you before me?")

```

Comment:
They aren't, there's two "pools" of phrases for poly, the speak list and the speech_buffer list.

The speech_buffer is 500 phrases long and is composed entirely of things poly has overheard. He cannot learn from himself.

The speak list is a list of 8 or 9 phrases that poly actively has on hand to quote from.

When poly says something, he picks a phrase from the speak list. Then, 80% of the time, he'll remove a random phrase from the speak list and replace it with another random phrase from the speech_buffer list. As the round progresses any phrases that are in the speak list but not in the speech_buffer list are extincted, and don't survive to the next round. Even if one of these initial phrases DID survive to round end/poly death, poly only remembers speech_buffer, so they're forever locked to conditionals like this (unless some chucklefuck teaches them to poly manually).